src/libs/Pusher/pusher.js

```diff
@@ -358,6 +360,46 @@ function reconnect() {
     socket.connect();
 }
 
+/**
+ * Abstraction around subscribing to private user channel events. Handles all logs and errors automatically.
+ *
+ * @param {String} eventName
+ * @param {String} accountID
+ * @param {Function} onEvent
+ * @param {Boolean} isChunked
+ */
+function subscribeToPrivateUserChannelEvent(eventName, accountID, onEvent, isChunked = false) {
+    const pusherChannelName = `private-encrypted-user-accountID-${accountID}${CONFIG.PUSHER.SUFFIX}`;
+
+    /**
+     * @param {Object} pushJSON
+     */
+    function logPusherEvent(pushJSON) {
+        Log.info(`[Report] Handled ${eventName} event sent by Pusher`, false, pushJSON);
+    }
+
+    function onPusherResubscribeToPrivateUserChannel() {
+        NetworkConnection.triggerReconnectionCallbacks('Pusher re-subscribed to private user channel');
```

Comment:
**opinion (non-blocking):** It stuck out to me that the Pusher library is now triggering and aware of the reconnection callbacks. Feels like it now has extra responsibility it should not have. I would maybe put this into something like a `PusherUtils` or just have it in `User` eventually (after the usages in `Report.js` are cleaned up). But not sure we need to do this now.

Reply:
I like putting it into `PusherUtils` :+1:

Reply:
Moved!